src/errcodes.h

```diff
@@ -0,0 +1,27 @@
+#ifndef YOBEMAG_ERRCODES_H
+#define YOBEMAG_ERRCODES_H
+
+typedef enum ErrorCode {
+    /**
+     * @brief Signal successful completion
+     */
+    ERR_SUCCESS,
```

Comment:
I think that the naming is not quite fitting here 🤔 
ERR_SUCCESS is basically an oximoron
I think that not using a prefix could be better here

Reply:
Well `ERR_` is just the prefix. I've seen this used in the implementation of an IP stack as well. 
![image](https://user-images.githubusercontent.com/20692341/196006641-ba1a3424-5569-48c9-a279-2889774a5ff6.png)
Do you have another prefix?

Reply:
Hmm I see, kinda funny that this is commonly done 😂
But you're right, it should have a prefix of some sort.
Maybe we could readjust a bit with the issue that we discussed in #14 about error propagation. If we put the the alternate error handling in place, we could use these as error codes only. That is, we do not need to tell the caller about success since no exit means success in the scenario I described in #14 🤔  Then ERR_ would be fine again